src/stm32/stm32f0_i2c.c

```diff
@@ -146,13 +146,24 @@ i2c_setup(uint32_t bus, uint32_t rate, uint8_t addr)
         enable_pclock((uint32_t)i2c);
         gpio_peripheral(ii->scl_pin, ii->function | GPIO_OPEN_DRAIN, 1);
         gpio_peripheral(ii->sda_pin, ii->function | GPIO_OPEN_DRAIN, 1);
-
-        // Set 100Khz frequency and enable
-        i2c->TIMINGR = ((0xB << I2C_TIMINGR_PRESC_Pos)
-                        | (0x13 << I2C_TIMINGR_SCLL_Pos)
-                        | (0xF << I2C_TIMINGR_SCLH_Pos)
-                        | (0x2 << I2C_TIMINGR_SDADEL_Pos)
-                        | (0x4 << I2C_TIMINGR_SCLDEL_Pos));
+        uint32_t pclk = get_pclock_frequency((uint32_t)i2c);
+        uint32_t presc = 1;
+        uint32_t scaled_pclk = pclk / (presc + 1);
+        // Minimal pclk 48Mhz stm32f0
+        while (scaled_pclk / rate > 255) {
+            presc += 1;
+            scaled_pclk = pclk / (presc + 1);
+        }
+        uint32_t total_cycles = scaled_pclk / rate;
+        uint32_t sdadel = total_cycles / 10;
+        uint32_t scldel = total_cycles / 10 - 1;
+        uint32_t scll = total_cycles * 4 / 10 - 1;
```

Comment:
Because we can't reuse values direct values, they are too low, for high-frequency peripheral clocks.
Like we have 4 bit presc and pclk one of 48Mhz, 64Mhz (stm32g0), 100Mhz (stm32h7/f7), 400Mhz (stm32g4/stm32L4).

Old values adds up like (4+1) + 2 + (15+1) + (19+1) = 43
 48_000_000 / 100_000 / 0xB (11) = 43
 
 There is no room to change presc to match anything above 64Mhz.
 64_000_000 / 100_000 / 15 = 42.66

As a consequence of the hacky formula here, it just changed.
_Looks like I messed up scldel/sdadel, because 255 / 10 > 15, I will fix that_

I do not see a way to match existing timings precisely, but indeed - I can try "be close" to them.